WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewModel.swift

```diff
@@ -251,6 +251,25 @@ extension ProductFormViewModel {
     func updateVariationAttributes(_ attributes: [ProductVariationAttribute]) {
         // no-op
     }
+
+    /// Updates the original product variations(and attributes).
+    /// This is needed because variations and attributes, remote updates, happen outside this view model and wee need a way to sync our original product.
+    ///
+    func updateProductVariations(from newProduct: Product) {
+        let newOriginalProduct = EditableProductModel(product: newProduct)
+
+        // If the product doesn't have any pending changes, we can safely override the original product
+        guard hasUnsavedChanges() else {
+            return resetProduct(newOriginalProduct)
+        }
+
+        // If the product has pending changes, we need to override the `originalProduct` first and the `living product` later with a saved copy.
+        // This is because, overriding `originalProduct` also overrides the `living product`.
+        let productWithChanges = EditableProductModel(product: product.product.copy(attributes: newProduct.attributes,
+                                                                                    variations: newProduct.variations))
+        resetProduct(newOriginalProduct)
```

Comment:
nit: just wondering, if there are unsaved changes on the product, does it make sense to make the original product based on `originalProduct` so that we call `resetProduct(EditableProductModel(product: originalProduct.product.copy(attributes:...variations...)))` here?

It might not make a difference UI-wise, since `originalProduct` is mostly used for showing the "Update" CTA and there are already outstanding changes from the new attributes and variations. Just thought it might be more consistent keeping the `originalProduct` 😄

Reply:
Nice, I think it does not make a difference in practice but it does read conceptually better!
Thanks for pointing this out!